### PR TITLE
Some BugFix following python3 port

### PR DESCRIPTION
**Pull request by DodoLaSaumure**
ops is filter type in python3 , convert to list

### CODE REVIEW

**Comment by Harvie** [comment]
Will this work in both python2 and python3? We still have python2 users...

**Comment by DodoLaSaumure** [comment]
@Harvie yes it works both on python2 and python3, just tested.

**Comment by DodoLaSaumure** [comment]
commit 25fb511 should correct #1314 https://github.com/vlachoudis/bCNC/issues/1314

**Comment by DodoLaSaumure** [comment]
Commit 2e41fc5 & a672e1c : 
python3 pickler does only accepts ByteIO, not StringIO, and I did not find anyway to convert Block dump tuples to bytearray. I found it more easy and more pythonic to wrap data in json objects.

**Comment by DodoLaSaumure** [comment]
all commits of this branch tested on both python2 & python3

**Comment by Harvie** [comment]
`import json` is internal part of python or does it need to be installed as bCNC dependency?

File: bCNC/lib/tkExtra.py [hunk 1]
```diff
@@ -3075,13 +3076,18 @@ def afterHide(self):
 	# Public methods
 	# ----------------------------------------------------------------------
 	def get(self, first=None, last=None):
+		PY2 = sys.version_info[0] == 2
```
Comment (Harvie) [inline bCNC/lib/tkExtra.py]: i would prefer python version checks not being used in code. is there another way to solve this? like better use of isinstance() ?

**Comment by DodoLaSaumure** [comment]
> `import json` is internal part of python or does it need to be installed as bCNC dependency?
json is internal part of python3
https://docs.python.org/fr/3/library/json.html
and python2
https://docs.python.org/fr/2.7/library/json.html

File: bCNC/lib/tkExtra.py (hunk 1 above)
Comment (DodoLaSaumure) [reply bCNC/lib/tkExtra.py]: Thanks @Harvie checkout commit 723bb86 please

**Comment by Harvie** [comment]
OK thanks, seems resonable... Do you have something more going on, or should i merge?

**Comment by DodoLaSaumure** [comment]
Thank you @Harvie Please merge !! And all the best for this end of year !!

**Comment by Harvie** [comment]
Copypasting now works for me on both py2 and py3. Thank you very much!

**Comment by Harvie** [comment]
@DodoLaSaumure can you please take a look at this? #1336
There seems to be some problem with json clipboard...

**Pull request merged**